vault/expiration.go

```diff
@@ -422,6 +454,31 @@ func (m *ExpirationManager) Restore(errorFunc func()) (retErr error) {
 	m.restoreModeLock.Unlock()
 
 	m.logger.Info("lease restore complete")
+
+	if !m.SHA1HashedLeasesCleared {
+		// At this time all the leases are loaded. The restoration of leases would
+		// have also computed if were any leases that had either its ID or
+		// secondary index obfuscated using SHA1 hash instead of SHA2-256 HMAC.
+		// Persist the result of that computation across reboots.
+		expMetadata := &ExpirationMetadata{}
+
+		if atomic.LoadUint32(&m.SHA1HashedLeasesPresent) == 0 {
```

Comment:
Would it be possible to combine/use a one of `SHA1HashedLeasesCleared` and `SHA1HashedLeasesPresent` since they are used to track the same state?

Reply:
One is used to persist the state and the other is used to compute the state. I am not sure if we can combine the two.